packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts

```diff
@@ -120,13 +125,18 @@ export class SagemakerTrainTask implements ec2.IConnectable, sfn.IStepFunctionsT
         };
 
         // set the sagemaker role or create new one
-        this.role = props.role || new iam.Role(scope, 'SagemakerRole', {
+        this.role = props.role || new iam.Role(scope, 'SagemakerTrainRole', {
             assumedBy: new iam.ServicePrincipal('sagemaker.amazonaws.com'),
             managedPolicies: [
                 iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSageMakerFullAccess')
```

Comment:
AppSec is not too happy about this. See https://github.com/awslabs/aws-cdk/pull/2991.

Not making you change this in this PR, since we have another one for it. But so you are aware.

Reply:
Sure thing